Automate_EDA_with_DataExplorer.Rmd

```diff
@@ -0,0 +1,174 @@
+
+
+* For example, the plot_bar function plots the distribution only for the discrete variables.
+```{r}
+##`Life` distribution of all discrete variables
+plot_bar(sleep1, with = "Life",title="Bar distribution in response to Life span")
+```
+
+* The plot_histogram function plots the distribution only for the continuous variables.
+
+
+```{r}
+plot_histogram(sleep1,title= "Histogram Distribution of sleep data")
+```
+
+
+* The plot_qq function plots the qq-plot only for the continuous variables.
+
+
+````{r,message=FALSE}
+##View quantile-quantile plot of all continuous variables by feature `Life`
+plot_qq(sleep1,by='Life',title="QQ-plot in response to Life span")
+```
+
+### 3 Correlation Analysis
+
+* DataExplorer provides correlation heatmap plot for all non-missing features. The heatmap can also be set to plot for continuous or discrete variables only.
+
+````{r}
+plot_correlation(na.omit(sleep1),title="Correlation heatmap of the sleep data")
+```
+
+
+````{r}
+plot_correlation(na.omit(sleep1),type = "c",,title="Correlation heatmap of the sleep data, continuous only")
+```
+* For the discrete variable, the heatmap automatically done one hot encoding that provides further insights.
+````{r}
+plot_correlation(na.omit(sleep1),type = "d",,title="Correlation heatmap of the sleep data,discrete only")
+```
+
+## 5. Feature Engineering
+
+* Feature engineering is often needed in the data analysis process to transform data into better representative features. DataExplorer provides mutiple functions for feature engineering including missing value filling, sparse categories grouping, one hot encoding and feature transformation.
+
+* The set_missing function can fill both the discrete and continuous variable with designated values in one line of code.
+
+### 1 Missing value
+````{r}
+final_sleep <- set_missing(sleep1, list(0L, "unknown"))
+plot_missing(final_sleep)
+```
+
+* One hot encoding allows the categorical data to be more expressive. It can be done by the dummify function.
+
+### 2 Dummy variable
+````{r}
+plot_str(
+  list(
+    "original" = final_sleep,
+    "dummified" = dummify(final_sleep)
+  )
+)
+```
+
+## 5. Data Report
+
+* All the summary statistic and visualization plots of the data set can be organized into a data report in 1 step. The report automatically generates most visualization plots above. It is indeed a rough data profile but very useful for initial analysis and user-friendly for beginners 
+
+````{r}
+#create_report(final_sleep)
+````
+
+![Data report sample](resources/Automate_EDA_with_DataExplorer/unknown.png)
```

Comment:
Update folder file name to all lower case to match .rmd. Also, the filename in the folder is '**U**nknown.png', not '**u**nknown.png', would update one of these to make the names consistent.